internal/action/actions.go

```diff
@@ -482,6 +483,20 @@ func (h *BufPane) CursorEnd() bool {
 	return true
 }
 
+// CursorRatio moves the cursor to a given ratio of the buffer.
+// For example, if ratio is 0.5, the cursor is moved to the center line.
+func (h *BufPane) CursorRatio(ratio float64) bool {
+	if ratio < 0 || ratio > 1 {
+		return false
+	}
+	targetLocation := buffer.Loc{0, int(math.Round(float64(h.Buf.End().Y) * ratio))}
+	h.Cursor.Deselect(true)
+	h.Cursor.Loc = targetLocation
```

Comment:
Please note that for goto (and generally for cases where the new location may be far away from the old location) you should use `h.GotoLoc()` rather than just set the new location and relocate. See #2628.